sdk/src/AbstractMain.cpp

```diff
@@ -90,7 +90,8 @@ int configureScratchPath(
             ".")
 
       // Drop privileges to the server  user.
-      error = system::posix::temporarilyDropPrivileges(in_serverUser);
+      const Optional<system::GidType> in_group;
+      error = system::posix::temporarilyDropPrivileges(in_serverUser, in_group);
```

Comment:
Did you just need an empty Optional here?  Would something like the following compile? Or 
```suggestion
      error = system::posix::temporarilyDropPrivileges(in_serverUser, {});
```

```suggestion
      error = system::posix::temporarilyDropPrivileges(in_serverUser, Optional<system::GidType>());
```

If not, you could name it `empty`:
```suggestion
      const Optional<system::GidType> noGroup;
      error = system::posix::temporarilyDropPrivileges(in_serverUser, noGroup);
```

On first read, it's confusing that we're declaring a group and never initialising it.